### DISCUSSION TITLE
How to release origin from cors in rabbitmq-tracing plugin

### DISCUSSION THREAD

**Discussion by Andersonfrfilho** [rabbitmq/rabbitmq-server#6689] [Other]
Hey guys, I hope you're doing well, I'm creating a rabbitmq mapper, and it made sense for me to track messages where they come from and where they go, it seems to me that the best way would be to use the rabbitmq-tracing plugin, but when I go create a tracing via the axios to request:

```js
const response = await axios.put(
    'http://localhost:15672/api/traces/node/rabbit%40rabbit-mapper-host/%2F/project-nextjs-producer-trace',
    {
        'vhost': '/',
        'name': 'project-nextjs-producer-trace',
        'format': 'json',
        'tracer_connection_username': 'project-nextjs-producer',
        'tracer_connection_password': 'guest',
        'pattern': '#',
        'node': 'rabbit@rabbit-mapper-host'
    },
    {
        headers: {
            'Accept': '*/*',
            'Accept-Language': 'pt-BR,pt;q=0.9,en-US;q=0.8,en;q=0.7',
            'Authorization': 'Basic Z3Vlc3Q6Z3Vlc3Q=',
        }
    }
);
```

front-end, I am blocked by the cors configuration. 
* in the rabbitmq HTTP requests I solved the cors problem through the file: etc/rabbitmq/rabbitmq.conf

**rabbitmq.conf**

```conf
## DEFAULT SETTINGS ARE NOT MEANT TO BE TAKEN STRAIGHT INTO PRODUCTION
## see https://www.rabbitmq.com/configure.html for further information
## on configuring RabbitMQ

## allow access to the guest user from anywhere on the network
## https://www.rabbitmq.com/access-control.html#loopback-users
## https://www.rabbitmq.com/production-checklist.html#users
loopback_users.guest = false

## Send all logs to stdout/TTY. Necessary to see logs when running via
## a container
log.console = true

management.cors.allow_origins.1 = *
```

but, how would solve the [rabbitmq-tracing](https://github.com/rabbitmq/rabbitmq-tracing) plugin cors problem ?
or would you have something better to use than the tracing plugin for this purpose ?

**Comment by michaelklishin** [comment]
Please use GitHub Discussions and not issues for questions, Discussions have been around for over a year now.

**Comment by michaelklishin** [comment]
`rabbitmq-tracing` is a plugin that's usually only used in development. So it does not have support for all the CORS features for its HTTP API endpoint(s). So the plugin has to be made aware of the CORS settings configured for the management plugin.

**Reply by Andersonfrfilho** [reply]
Would there be any other way to track these messages? In production environments?
